go/adbc/driver/databricks/record_builder.go

```diff
@@ -0,0 +1,338 @@
+package databricks
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/apache/arrow-go/v18/arrow"
+	"github.com/apache/arrow-go/v18/arrow/array"
+	"github.com/apache/arrow-go/v18/arrow/memory"
+)
+
+// ColumnBuilder is an interface for building Arrow arrays
+type ColumnBuilder interface {
+	Append(value interface{}) error
+	Builder() array.Builder
+}
+
+// StringColumnBuilder handles string values
+type StringColumnBuilder struct {
+	builder *array.StringBuilder
+}
+
+func NewStringColumnBuilder(mem memory.Allocator) *StringColumnBuilder {
+	return &StringColumnBuilder{
+		builder: array.NewStringBuilder(mem),
+	}
+}
```

Comment:
Since these are called only once, you could probably inline the 3-liners in the switch statement from where these are being called.